### PR TITLE
Resolve GSSAPI rolvaliduntil FIXME

### PR DESCRIPTION
**Pull request by danielgustafsson**
The FIXME was added during the PostgreSQL 9.0 merge as a note to self to double-check the logic for GSS password expiry times. Verified that the code is indeed in sync, and removed the FIXME leaving most of the comment in place as a reminder to keep the two functions synchronized (consolidating them would be neat but it would also introduce risk in a very sensitive codepath so for now I think keeping them separate is the right thing to do).

Also cleaned up a few style issues and moved to the same syscache convenience macro that the password check uses.

### CODE REVIEW

**Comment by ivannovick** [comment]
interesting because we had some complaints on whether this works properly in the past.  Glad to see we have verified it

File: src/backend/libpq/auth.c [hunk 1]
```diff
@@ -1125,18 +1127,16 @@ pg_GSS_error(int severity, char *errmsg, OM_uint32 maj_stat, OM_uint32 min_stat)
...
 
-/* Check to see if the password of a user is valid 
- * (using the validuntil attribute associated with the pg_role)
- * for GSSAPI based authentication.
+/*
+ * Check to see if the password of a user is valid (using the validuntil
+ * attribute associated with the pg_role) for GSSAPI based authentication.
+ *
+ * This logic is copied from hashed_passwd_verify(), so we need to ensure
+ * this these functions doesn't fall out of sync.
```
Comment (hlinnaka) [inline src/backend/libpq/auth.c]: spurious "this"

Comment (hlinnaka) [reply src/backend/libpq/auth.c]: "doesn't" -> "don't"

Comment (danielgustafsson) [reply src/backend/libpq/auth.c]: Fixed and rebased, thanks!

**Comment by danielgustafsson** [comment]
Rebased and pushed to master.

**Pull request closed without merging**